### PR TITLE
ffb_2d gpu

### PR DESCRIPTION
**Pull request by j-c-c**

### CODE REVIEW

File: src/aspire/basis/ffb_2d.py [hunk 1]
```diff
@@ -147,22 +150,19 @@ def _evaluate(self, v):
...
                 pf[:, 2 * n_theta - ell, :] = -pf_ell.conjugate()
 
-            ind = ind + np.size(idx)
-            ind_pos = ind_pos + 2 * self.k_max[ell]
+            ind = ind + idx.size
+            ind_pos = ind_pos + 2 * xp.array(self.k_max[ell])
 
         # 1D inverse FFT in the degree of polar angle
-        pf = 2 * pi * xp.asnumpy(fft.ifft(xp.asarray(pf), axis=1))
+        pf = 2 * xp.pi * fft.ifft(xp.asarray(pf), axis=1)
```
Comment (garrettwrong) [inline src/aspire/basis/ffb_2d.py]: does this still need to be asarray'd ?

**Review by garrettwrong — CHANGES_REQUESTED** [review]
Awesome thanks, just one review question.

Otherwise if you confirmed ran in both numpy and cupy configs we should be good to merge into `hack24`.

File: src/aspire/basis/ffb_2d.py (hunk 1 above)
Comment (j-c-c) [reply src/aspire/basis/ffb_2d.py]: Oops, it does not. Thanks for the catch!

**Review by garrettwrong — APPROVED** [review]
Great.  Merge when ready. Thanks!

**Pull request merged**